.github/workflows/check-which-tests-to-run.yaml

```diff
@@ -0,0 +1,71 @@
+name: Check which tests to run
+
+on:
+  workflow_call:
+    outputs:
+      weave_query_tests:
+        value: ${{ jobs.check.outputs.weave_query_tests }}
+      weave_js_tests:
+        value: ${{ jobs.check.outputs.weave_js_tests }}
+      core_integration_tests:
+        value: ${{ jobs.check.outputs.core_integration_tests }}
+
+env:
+  WEAVE_QUERY_PATHS: 'weave_query/'
+  WEAVE_JS_PATHS: 'weave-js/'
+  CORE_INTEGRATION_PATHS: 'weave-js/ weave_query/'
+  # Everything else is implicitly trace server
```

Comment:
do we need 1 more for trace server?

Reply:
Once we fully isolate trace sdk and trace server, we could further split this